discretisedfield/plotting/mpl.py

```diff
@@ -0,0 +1,873 @@
+                                    cmap=vector_cmap, clim=vector_clim,)
+
+            if filename is not None:
+                plt.savefig(filename, bbox_inches='tight', pad_inches=0.02)
+            ```
+
+            Therefore, to understand the meaning of the arguments which can be
+            passed to this method, please refer to
+            ``discretisedfield.plotting.Mpl.scalar`` and
+            ``discretisedfield.plotting.Mpl.vector`` documentation.
+
+        Example
+        -------
+        .. plot:: :context: close-figs
+
+            1. Visualising the field using ``matplotlib``.
+
+            >>> import discretisedfield as df
+            ...
+            >>> p1 = (0, 0, 0)
+            >>> p2 = (100, 100, 100)
+            >>> n = (10, 10, 10)
+            >>> mesh = df.Mesh(p1=p1, p2=p2, n=n)
+            >>> field = df.Field(mesh, dim=3, value=(1, 2, 0))
+            >>> field.plane(z=50, n=(5, 5)).mpl.plot_simplified()
+
+        .. seealso::
+
+            :py:func:`~discretisedfield.plotting.Mpl.scalar`
+            :py:func:`~discretisedfield.plotting.Mpl.vector`
+
+        """
+        ax = self._setup_axes(ax, figsize)
+
+        multiplier = (uu.si_max_multiplier(self.data.mesh.region.edges)
+                      if multiplier is None else None)
+
+        # Define defaults and update with user-passed dictionaries.
+        scalar_kwargs = {}
+        vector_kwargs = {'use_color': False, 'colorbar': False}
+        scalar_kwargs.update(scalar_args)
+        vector_kwargs.update(vector_args)
+
+        # Set up default scalar and vector fields.
+        # TODO: Do we allow user to specify what scalar and vector fields are? Did we have that before?
+        if self.data.dim == 1:
+            scalar_field = self.data
+            vector_field = None
+
+        elif self.data.dim == 2:
+            scalar_field = self.data
+            vector_field = self.data
+
+        else:
+            vector_field = self.data
+            scalar_field = getattr(self.data, self.planeaxis)
+            scalar_kwargs['colorbar_label'] = scalar_kwargs.get('colorbar_label',
+                                                                f'{self.planeaxis}-component')
+
+        # Since this is done in all three branches of if statement, I expose it here.
+        # TODO: Doule check if I miss something.
+        # TODO: What is the norm here if dim == 1?
+        scalar_kwargs['filter_field'] = scalar_kwargs.get('filter_field',
+                                                          self.data.norm)
+
+        scalar_field.mpl.scalar(ax=ax, multiplier=multiplier, **scalar_kwargs)
+        if vector_field is not None:
+            vector_field.mpl.vector(ax=ax, multiplier=multiplier, **vector_args)
+
+        self._axis_labels(ax, multiplier)
+
+        if filename is not None:
+            plt.savefig(filename, bbox_inches='tight', pad_inches=0.02)
+
+    def plot(self,
```

Comment:
Should be removed once we decide what exactly to support in `plot_simplified`. (Code is for now only kept for reference.)